### PR TITLE
Add support for query params

### PR DESCRIPTION
**Pull request by hamza3202**

### CODE REVIEW

**Comment by stakater-user** [comment]
@hamza3202 Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins.release.stakater.com/job/stakater/job/GitWebhookProxy/job/PR-63/1/) has Failed!

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -101,7 +101,9 @@ func (p *Proxy) redirect(hook *providers.Hook, path string) (*http.Response, err
 }
 
 func (p *Proxy) proxyRequest(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
-	log.Printf("Proxying Request from '%s', to upstream '%s'\n", r.URL, p.upstreamURL+r.URL.Path)
+	redirectURL := p.upstreamURL + r.URL.Path + "?" + r.URL.RawQuery
```
Comment (waseem-h) [inline pkg/proxy/proxy.go]: What if we don't have anything in the raw query? It will append `?` nonetheless

**Comment by stakater-user** [comment]
@hamza3202 Image is available for testing. ``docker pull stakater/gitwebhookproxy:SNAPSHOT-PR-63-2``

**Comment by jasonliu747** [comment]
Tested with stakater/gitwebhookproxy:SNAPSHOT-PR-63-2 and got some errors:
`2019/04/26 14:04:50 Github payload unmarshaling failed for Push event: json: cannot unmarshal string into Go struct field .created_at of type int64
2019/04/26 14:04:50 Now trying to unmarshal for pull request event
2019/04/26 14:04:50 Event type is not supported: ping
2019/04/26 14:04:50 Incoming request from user: 
2019/04/26 14:04:50 Ignoring request for user: 
`

**Comment by jasonliu747** [comment]
@hamza3202 `ping` event was supported, but now the response will be `Ignoring request for user:`

**Comment by hamza3202** [comment]
@jasonliu747  what do you mean?

**Comment by jasonliu747** [comment]
> @jasonliu747 what do you mean?

image
this was what I got with `stakater/gitwebhookproxy:v0.2.63`

image
and this one with `stakater/gitwebhookproxy:SNAPSHOT-PR-63-2`

**Comment by hamza3202** [comment]
@jasonliu747 can you share the request data? I think the request body doesn't specify the committer. Also only Pull request and push events are supported

**Comment by jasonliu747** [comment]
@hamza3202 no, it didn't specify the committer. I just wanna let you know, with the `stakater/gitwebhookproxy:v0.2.63` image, I won't get an ingnoring error in response.

**Comment by jasonliu747** [comment]
Another feedback. I got the following error when trying to enter the container by using `docker exec -it <CONTAINER ID>  /bin/bash` with `stakater/gitwebhookproxy:SNAPSHOT-PR-63-2` image.

**_OCI runtime exec failed: exec failed: container_linux.go:345: starting container process caused "exec: \"/bin/bash\": stat /bin/bash: no such file or directory": unknown_**

**Comment by hamza3202** [comment]
@jasonliu747  
- bash is not available in the GitWebhookProxy image and hence you are getting the above error. It was not available in the previous version either.
- I am not able to generate the scenario you have mentioned above. Also the current log is the correct one.

Over all, the PR is doing the task assigned correctly so I believe this PR should be merged and other issues should be discussed in separate issues.

@kahootali please review.

**Comment by jasonliu747** [comment]
@hamza3202 

- It is available in the previous version.
image

- All you need to do is adding a new webhook in a Github project, and it'll create a ping event. That's how I generate the scenario.

**Pull request merged**